src/Groups/GroupDetails/GroupDetailsContainer.js

```diff
@@ -32,15 +34,34 @@ const GroupDetailsContainer = ({
   onClose,
 }) => {
   const groupId = match.params.id;
-  const group = get(resources, ['groupDetails', 'records', '0']);
+  const [groupData, setGroupData] = useState({ isLoading: true });
   const showToast = useShowToast();
 
   useEffect(
     () => {
-      mutator.groupDetails.reset();
-      mutator.currentFiscalYears.reset();
-      mutator.groupDetails.GET();
-      mutator.currentFiscalYears.GET();
+      const groupDetailsPromise = mutator.groupDetails.GET();
+      const currentFiscalYearPromise = mutator.currentFiscalYears.GET();
+
+      Promise.all([groupDetailsPromise, currentFiscalYearPromise])
+        .then(responses => {
+          const groupDetails = responses[0];
+          const currentFiscalYears = responses[1];
+
+          setGroupData(prevGroupData => ({
+            ...prevGroupData,
+            groupDetails,
+            currentFiscalYears,
+          }));
+
+          return getGroupSummary(mutator.groupSummaries, groupId, get(currentFiscalYears, [0, 'id']));
+        })
+        .then(groupSummary => {
+          setGroupData(prevGroupData => ({
+            ...prevGroupData,
```

Comment:
what is the reason to keep previous data?

Reply:
because it updates only part of this data